### ISSUE TITLE
This feature is not available for your account - with zend fatal error

### ISSUE THREAD

**Issue by ashleyse2016** [SpamExperts/cpanel-addon#37]
### Version information

```
Controlpanel: Cpanel v11.68.0.12
PHP version: 5.6.30
Addon version: 3.0.96549
```

### Steps to replicate

 1. Login to WHM server
 2. Navigate to list domains
 3. On screen error

1. navigate to cPanel user
2. Click Professional spam filter
3. On screen indicates `This feature is not available for your account`

There is no addon log when this happens. 

### Actual result

```
Fatal error: Uncaught exception 'Zend_Session_Exception' with message 'You must explicitly start the session with Zend_Session::start() when session options are set to strict.' in /usr/local/prospamfilter/library/Zend/Session.php:436 Stack trace: #0 /usr/local/prospamfilter/library/Zend/Session/Namespace.php(143): Zend_Session::start(true) #1 /usr/local/prospamfilter/library/Zend/Controller/Action/Helper/FlashMessenger.php(82): Zend_Session_Namespace->__construct('FlashMessenger') #2 /usr/local/prospamfilter/interfaces/admin.php(66): Zend_Controller_Action_Helper_FlashMessenger->__construct() #3 /usr/local/prospamfilter/frontend/whm/prospamfilter.php(35): require_once('/usr/local/pros...') #4 {main} thrown in /usr/local/prospamfilter/library/Zend/Session.php on line 436
```

### Expected result

List all domains, and also cPanel user level list domains should work. 

### Other notes

Added logging from client

```
[2017-11-07 05:05:59 -0500] info [xml-api] Permission denied: You do not have the required privileges to run “verify_user_has_feature”. [verify_user_has_feature] version [1].
Cannot get grandparent of grandparent process name.
Cannot get grandparent of grandparent process name.
Cannot get grandparent of grandparent process name.
Cannot get grandparent of grandparent process name.
Cannot get grandparent of grandparent process name.
Cannot get grandparent of grandparent process name.
Cannot get grandparent of grandparent process name.
Cannot get grandparent of grandparent process name.
Cannot get grandparent of grandparent process name.
[2017-11-07 05:06:40 -0500] info [xml-api] Permission denied: You do not have the required privileges to run “verify_user_has_feature”. [verify_user_has_feature] version [1].
[2017-11-07 05:08:49 -0500] warn [pureauth] Failed to call hulk pre_login for anonymous@REDACTED  (system) at bin/pureauth.pl line 135.
[07-Nov-2017 10:25:44 UTC] PHP Fatal error: Uncaught exception 'Zend_Session_Exception' with message 'You must explicitly start the session with Zend_Session::start() when session options are set to strict.' in /usr/local/prospamfilter/library/Zend/Session.php:436
Stack trace:
#0 /usr/local/prospamfilter/library/Zend/Session/Namespace.php(143): Zend_Session::start(true)
#1 /usr/local/prospamfilter/library/Zend/Controller/Action/Helper/FlashMessenger.php(82): Zend_Session_Namespace->__construct('FlashMessenger')
#2 /usr/local/prospamfilter/interfaces/admin.php(66): Zend_Controller_Action_Helper_FlashMessenger->__construct()
#3 /usr/local/prospamfilter/frontend/whm/prospamfilter.php(35): require_once('/usr/local/pros...')
#4 {main}
thrown in /usr/local/prospamfilter/library/Zend/Session.php on line 436
[2017-11-07 05:25:44 -0500] warn [whostmgrd] The subprocess (/usr/local/cpanel/whostmgr/docroot/cgi/addon_prospamfilter.cgi) exited with an error: The subprocess reported error number 255 when it ended. at /usr/local/cpanel/Cpanel/Server/Handlers/SubProcess.pm line 251.
Cpanel::Server::Handlers::SubProcess::_report_subprocess_errors(Cpanel::Server::Handlers::SubProcess=HASH(0x1fbdb38)) called at /usr/local/cpanel/Cpanel/Server/Handlers/SubProcess.pm line 109
Cpanel::Server::Handlers::SubProcess::handler(Cpanel::Server::Handlers::SubProcess=HASH(0x1fbdb38), __CPANEL_HIDDEN__, __CPANEL_HIDDEN__, __CPANEL_HIDDEN__, 822722, __CPANEL_HIDDEN__, IO::Handle=GLOB(0x1fbda18), __CPANEL_HIDDEN__, ...) called at cpsrvd.pl line 6511
cpanel::cpsrvd::cgiHandler(__CPANEL_HIDDEN__, __CPANEL_HIDDEN__, __CPANEL_HIDDEN__, __CPANEL_HIDDEN__) called at cpsrvd.pl line 5981
cpanel::cpsrvd::dodoc_whostmgrd() called at cpsrvd.pl line 1651
cpanel::cpsrvd::dodoc(HASH(0x14efd00)) called at cpsrvd.pl line 1429
cpanel::cpsrvd::handle_one_connection(7) called at cpsrvd.pl line 877
cpanel::cpsrvd::script() called at cpsrvd.pl line 324
warn [whostmgrd] The subprocess (/usr/local/cpanel/whostmgr/docroot/cgi/addon_prospamfilter.cgi) exited with an error: The subprocess reported error number 255 when it ended.
PHP Fatal error: Call to a member function whm_api() on null in /usr/local/prospamfilter/library/SpamFilter/PanelSupport/Cpanel.php on line 1645
```

**Comment by for93t** [comment]
The cPanel user misses required WHM XML API permission, that's why there is an error appears in the log:
```
[2017-11-07 05:06:40 -0500] info [xml-api] Permission denied: You do not have the required privileges to run “verify_user_has_feature”. [verify_user_has_feature] version [1].
```

**Comment by Agoodfella** [comment]
The initial error cannot be reproduced by following the steps described in this ticket. 

Fix verified.

**Comment by ashleyse2016** [comment]
Reopeneing, as client tested the fix and continues to get issue:

```
Actual proposed fix:
--- /usr/local/prospamfilter/interfaces/admin.php.or 2017-11-16 00:52:46.761994528 -0500
+++ /usr/local/prospamfilter/interfaces/admin.php 2017-11-16 00:58:45.042993777 -0500
@@ -62,5 +62,7 @@ Zend_Layout::startMvc(array(
Zend_Controller_Front::run(ADDON_ROOT_FOLDER . '/application/controllers');

require_once 'Zend/Controller/Action/HelperBroker.php';
+Zend_Session::setOptions(array ("strict" => false));
Zend_Controller_Action_HelperBroker::addHelper(
new Zend_Controller_Action_Helper_FlashMessenger());
+Zend_Session::setOptions(array ("strict" => true));
```

However the error logs continue to show the issue:

```
[2017-11-16 00:56:07 -0500] info [xml-api] Permission denied: You do not have the required privileges to run âverify_user_has_featureâ
Cannot get grandparent of grandparent process name.
Cannot get grandparent of grandparent process name.
Cannot get grandparent of grandparent process name.
Cannot get grandparent of grandparent process name.
Cannot get grandparent of grandparent process name.
Cannot get grandparent of grandparent process name.
Cannot get grandparent of grandparent process name.
Cannot get grandparent of grandparent process name.
Cannot get grandparent of grandparent process name.
[2017-11-16 00:57:51 -0500] info [xml-api] Permission denied: You do not have the required privileges to run âverify_user_has_featureâ
Cannot get grandparent of grandparent process name.
Cannot get grandparent of grandparent process name.
Cannot get grandparent of grandparent process name.
Cannot get grandparent of grandparent process name.
Cannot get grandparent of grandparent process name.
Cannot get grandparent of grandparent process name.
Cannot get grandparent of grandparent process name.
Cannot get grandparent of grandparent process name.
Cannot get grandparent of grandparent process name.
[2017-11-16 00:58:26 -0500] info [xml-api] Permission denied: You do not have the required privileges to run âverify_user_has_featureâ
```

**Comment by alexkiro** [comment]
> Reopeneing, as client tested the fix and continues to get issue:

@ashleyse2016 we haven't released this fix in any build, what do you mean here?

**Comment by ashleyse2016** [comment]
The client tried the existing changeset https://github.com/SpamExperts/cpanel-addon/commit/ad47a0295b892c0f970c6c13200a45c5f177b7e2

**Comment by for93t** [comment]
The `Permission denied: ...` issue will be fixed in #35.
